### PR TITLE
MSSDK-1275 fix price rounding and add DKK in planHelper

### PR DESCRIPTION
**Pull request by kkarasek**
## Changes
- set grossPrice as offer's customerPriceIncludingTax and format using formatNumber() 
- changed grossPrice scope to global 
- updated subscription descriptions accordingly
- removed already formatted value in Price component
- fixed total amount formatting in ThankYouPage
- added correct DKK currency display in planHelper

offer_pricebox_rounded

thank_you_page_rounded

### CODE REVIEW

**Review by jkorczakcleeng — DISMISSED** [review]

File: src/components/OfferCheckoutCard/OfferCheckoutCard.js [hunk 1]
```diff
@@ -28,34 +28,30 @@ const OfferCheckoutCard = ({ isDataLoaded, t }) => {
     freePeriods,
     freeDays,
     expiresAt,
-    startTime
+    startTime,
+    customerPriceInclTax: grossPrice
```
Comment (iwonakulacz) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.js:32]: let's take price from order - it's our source of truth.
So lets keep 
```
const {
    priceBreakdown: { offerPrice },
    taxRate,
    country,
    currency
  } = useSelector(state => state.order.order);
  ```
Tax rate is used when we don't have a gross price (eg. trial), bo let's calculate it based on order - as it was

File: src/components/OfferCheckoutCard/OfferCheckoutCard.js [hunk 2]
```diff
@@ -28,34 +28,30 @@ const OfferCheckoutCard = ({ isDataLoaded, t }) => {
...
-  const {
-    priceBreakdown: { offerPrice },
-    taxRate,
-    country,
-    currency
-  } = useSelector(state => state.order.order);
+  const { country, currency } = useSelector(state => state.order.order);
   const offerType = offerId?.charAt(0);
   const currencySymbol = currencyFormat[currency];
+  const formattedGrossPrice = formatNumber(grossPrice);
```
Comment (iwonakulacz) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.js:37]: As we will go back to using an order price value, I would move the old `grossPrice` from l:42 to l:37. Then we will be able to keep the same variable name (`grossPrice`).
Then in the translations, we will have the same variables name, which is crucial for publishers that are already using the translations feature (so in l:41 we want to keep 'grossPrice' as a variable)

File: src/components/ThankYouPage/ThankYouPage.js [hunk 3]
```diff
@@ -31,6 +32,7 @@ const ThankYouPage = ({ onSuccess, t }) => {
   };
   const paymentMethodName = readablePaymentMethod[paymentMethod];
   const currencySymbol = currencyFormat[currency];
+  const formattedTotalAmount = formatNumber(totalAmount);
```
Comment (iwonakulacz) [inline src/components/ThankYouPage/ThankYouPage.js:35]: same here - we want to keep the same variable name in translation: totalAmount. You can rename it when getting from store and then here you may have something like: 
```
const totalAmount = formatNumber(totalAmountFromStore);
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.js (hunk 2 above)
Comment (iwonakulacz) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.js:37]: The root cause of the bug (6.99 -> 7 EUR) was caused by `formatNumber(offerPrice + taxRate * offerPrice)`
Screen Shot 2023-03-24 at 8 33 30 AM
This calculation is required to present properly the price for offers with trials (then, totalPrice = 0, and taxValue = 0. We need to present the final price WITH tax).
As a simple solution, I would recommend to: 
Ass util as here https://stackoverflow.com/a/11818658 and use it for `grossPrice`

**Comment by kkarasek** [comment]
### Amendments

- add formatTrialPrice and formatTrialTax functions 
- revert translation changes in CheckoutPriceBox component

### Description
customerPriceInclTax used to check whether grossPrice and taxValue should be rounded or truncated. Should work with currency exchange calculations alike.

The problem has been discussed with backend team, however the calculations provided by core does not work in case currency exchange.

**Review by jkorczakcleeng — APPROVED** [review]

File: src/components/CheckoutPriceBox/CheckoutPriceBox.js [hunk 4]
```diff
@@ -66,7 +67,7 @@ const CheckoutPriceBox = ({ t }) => {
               {!taxValue && taxRate && isCouponApplied && (
                 <p style={{ textDecoration: 'line-through' }}>
                   {currencyFormat[currency]}{' '}
-                  {formatNumber(taxRate * offerPrice)}
+                  {formatTrialTax(offerPrice, taxRate, customerPriceInclTax)}
```
Comment (iwonakulacz) [inline src/components/CheckoutPriceBox/CheckoutPriceBox.js:70]: rename: `formatTrialTax`, eg. `calculateTaxForFreeOffer`

File: src/util/formatTrialTax.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const formatTrialTax = (price, tax, incTax) => {
+  const taxValue = price * tax;
+  const isGrossAndTaxEqual = incTax === Number(price + tax * price).toFixed(2);
```
Comment (iwonakulacz) [inline src/util/formatTrialTax.js:3]: let's rename isGrossAndTaxEqual to something like `shouldRoundTaxValue`
+ 
`const truncatedTaxValue = (Math.trunc(taxValue * 100) / 100).toFixed(2)`;
return truncatedTaxValue;

File: src/util/formatTrialPrice.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const formatTrialPrice = (price, tax, incTax) => {
```
Comment (iwonakulacz) [inline src/util/formatTrialPrice.js:1]: - let's also rename `formatTrialPrice` to something like `calculateGrossPriceForFreeOffer`
- rename params to: netPrice, taxRate, customerPriceInclTax

File: src/util/formatTrialPrice.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  const grossPrice = price + tax * price;
+  const isGrossAndTaxEqual = incTax === Number(grossPrice).toFixed(2);
```
Comment (iwonakulacz) [inline src/util/formatTrialPrice.js:3]: `shouldRoundGrossPrice`

File: src/util/formatTrialPrice.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+  if (isGrossAndTaxEqual) {
+    return grossPrice.toFixed(2);
+  }
+  return (Math.trunc(grossPrice * 100) / 100).toFixed(2);
```
Comment (iwonakulacz) [inline src/util/formatTrialPrice.js:8]: // required to handle price edge cases

**Review by iwonakulacz — APPROVED** [review]

**Review by iwonakulacz — APPROVED** [review]

**Pull request merged**